phoenix-ios/phoenix-ios/views/compatibility/VibrationFeedback.swift

```diff
@@ -0,0 +1,36 @@
+import SwiftUI
+
+enum VibrationFeedbackType {
+	case error
+	
+	@available(iOS 17.0, *)
+	func convert() -> SensoryFeedback {
+		switch self {
+			case .error : return SensoryFeedback.error
+		}
+	}
+}
+
+struct VibrationFeedback<T>: ViewModifier where T: Equatable {
+	let type: VibrationFeedbackType
```

Comment:
When do these vibration happen? on my device (iphoneX, iOS 16.7.7) with haptic enabled in Settings > Sound & haptic, I don't get feedback when pressing PIN buttons, or when entering an invalid PIN.

Reply:
That particular feature is iOS 17 only. Although there are other ways to accomplish it on older iOS versions. I'll take a look.

Reply:
Done in https://github.com/ACINQ/phoenix/pull/560/commits/70cb899f3dce3cfc3498ae52997d84685a4d7a77